File: src/main/java/sortingalgorithms/bubblesort/java/BubbleSort.java
package sortingalgorithms.bubblesort.java;

import org.jetbrains.annotations.NotNull;
import sortingalgorithms.SortingAlgorithm;

import java.util.Arrays;

public class BubbleSort extends SortingAlgorithm {
    @Override
    public void sort(@NotNull int[] input) {
        int length = input.length;
        boolean isSorted;
        for (int i = 0;i< length;i++){
            isSorted = true;
            for (int j = 1;j< length -i;j++){
                if(input[j] < input[j-1]){
                    swap(input,j,j-1);
                    isSorted = false;
                }
            }
            if(isSorted) {
                Arrays.toString(input);
                return;
            }
        }
    }
}

File: src/main/java/sortingalgorithms/bubblesort/kotlin/BubbleSort.kt
package sortingalgorithms.bubblesort.kotlin

import sortingalgorithms.SortingAlgorithm
import java.util.*

class BubbleSort  : SortingAlgorithm(){
    override fun sort(input: Array<Int>) {
      val length = input.size
        var isSorted : Boolean
        for(i in 0 until length){
            isSorted = true
            for(j in 1 until  length -i){
                if(input[j] < input[j-1]){
                    swap(input,j,j-1)
                    isSorted = false
                }
            }
            if(isSorted){
                print(Arrays.toString(input))
                return
            }
        }
    }
}